src/param_spec.rs

```diff
@@ -612,6 +612,7 @@ pub trait ParamSpecType:
 }
 
 extern "C" {
+    #[no_mangle]
```

Comment:
It's unclear to me why this is actually required here but not for functions in `extern "C"` blocks and only on Windows/MSVC. The [nomicon](https://doc.rust-lang.org/nomicon/ffi.html#accessing-foreign-globals) also does not mention it.

A stackoverflow answer from @kornelski also mentions it though: https://stackoverflow.com/questions/45550387/how-can-i-access-a-c-global-variable-constant-in-rust-ffi

Reply:
I'm not sure if that code is necessary, or just happens to parse.

Reply:
Well here it fixes a linker error :) So some more digging is needed for what `rustc` is actually doing here.

Reply:
`#[no_mangle]` wasn't right fix. we need `#[link(name = "gobject-2.0")]` for the fix